packages/gatsby/src/query/index.ts

```diff
@@ -253,7 +253,7 @@ function createPageQueryJob(
   const { path, componentPath, context } = page
   if (_CFLAGS_.GATSBY_MAJOR === `4`) {
```

Comment:
can we remove this bit like in https://github.com/gatsbyjs/gatsby/pull/32896/files#diff-ef67aa343792980861ec389a60596e2a5dbf2994a7563c0fce855cd8eb3b1b52L254 as it becomes problematic when doing develop or build (can't remember)

We should filter here:
https://github.com/gatsbyjs/gatsby/pull/32896/files#diff-0d4a1161657d7b7e5edd5924370cf714c7777c18150ee67446dc972bfb369d42R102-R107 and in develop I had to change these bits https://github.com/gatsbyjs/gatsby/pull/32896/files#diff-d61539fe69ce8a4900be5d6476c6bfa63d4b18375d9859c099098e084cfd1841L203

Reply:
I will change the way we filter out queries but won't touch the `page-data` piece yet - it will be a part of a page engine PR.